platform/android/java/lib/src/org/godotengine/godot/input/GodotInputHandler.java

```diff
@@ -201,6 +226,67 @@ public void run() {
 				}
 			});
 			return true;
+		} else if ((event.getSource() & InputDevice.SOURCE_MOUSE) == InputDevice.SOURCE_MOUSE) {
+			if (event.getActionMasked() == MotionEvent.ACTION_HOVER_ENTER ||
+					event.getActionMasked() == MotionEvent.ACTION_HOVER_MOVE ||
+					event.getActionMasked() == MotionEvent.ACTION_HOVER_EXIT) {
+				final int x = Math.round(event.getX());
+				final int y = Math.round(event.getY());
+				final int type = event.getAction();
+				queueEvent(new Runnable() {
+					@Override
+					public void run() {
+						GodotLib.hover(type, x, y);
+					}
+				});
+				return true;
+			} else if (event.getAction() == MotionEvent.ACTION_BUTTON_PRESS) {
+				final int x = Math.round(event.getX());
+				final int y = Math.round(event.getY());
+				final int button = event.getActionButton();
+				queueEvent(new Runnable() {
+					@Override
+					public void run() {
+						GodotLib.mouse_pressed(button, x, y, true);
+					}
+				});
+				pressedMouseButtons[button - 1] = true;
+				return true;
+			} else if (event.getAction() == MotionEvent.ACTION_BUTTON_RELEASE) {
+				final int x = Math.round(event.getX());
+				final int y = Math.round(event.getY());
+				final int button = event.getActionButton();
+				queueEvent(new Runnable() {
+					@Override
+					public void run() {
+						GodotLib.mouse_pressed(button, x, y, false);
+					}
+				});
+				pressedMouseButtons[button - 1] = false;
+				return true;
+			} else if (event.getAction() == MotionEvent.ACTION_SCROLL) {
+				final int x = Math.round(event.getX());
+				final int y = Math.round(event.getY());
+
+				if (event.getAxisValue(MotionEvent.AXIS_VSCROLL) < 0.0f)
```

Comment:
What about horizontal scroll support?

Reply:
supported by wheel up and wheel down support by my testing

Reply:
I can't test it and I don't have any mouse that support it, and I didn't know that this is a thing, will add that